internal/util/ssa/patch.go

```diff
@@ -103,6 +112,9 @@ func Patch(ctx context.Context, c client.Client, fieldManager string, modified c
 		client.ForceOwnership,
 		client.FieldOwner(fieldManager),
 	}
+	if options.WithDryRun {
```

Comment:
We should never add something to the cache in l.136 if DryRun is set. Probably just make WithDryRun and WithCachingProxy mutually exclusive

Reply:
Great catch!